client/src/context.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+const Context = React.createContext();
+
+export class AuthProvider extends React.Component {
+  state = {
+    logged: false,
+    admin: false,
+  };
+
+  componentDidMount() {
+    fetch('/api/v1/auth')
+      .then((res) => {
+        if (res.ok) return res.json();
+        throw new Error();
+      })
+      .then((res) => {
+        this.setState({ logged: true, admin: res.role });
+      })
+      .catch(() => {
+        this.setState({ logged: false, admin: false });
+      });
+  }
+
+  render() {
+    const { logged, admin } = this.state;
+    return (
+      <Context.Provider
+        value={{
+          logged,
+          admin,
+        }}
+      >
+        {React.Children}
```

Comment:
```js
{ this.props.children }
```
you might need to use destructuring
and props validation
```js
  render() {
    const { isAuth, logged, admin } = this.state;
    const { children } = this.props;
    return (
      <Context.Provider
        value={{
          isAuth,
          logged,
          admin,
        }}
      >
        {children}
      </Context.Provider>
    );
  }
}
AuthProvider.propTypes = { children: PropTypes.node.isRequired };
```

Reply:
also you might need to export in a different way now
```js
export { AuthProvider, AuthConsumer };
```

Reply:
she can just
```js
export AuthProvider
```
named exports